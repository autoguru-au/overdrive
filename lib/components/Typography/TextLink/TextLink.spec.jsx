import React from 'react';
import { render } from '@testing-library/react';
import { TextLink } from '.';

describe('<TextLink />', () => {
	it('should not throw', () => {
		expect(() => render(<TextLink />)).not.toThrow();
	});

	it('should match the snapshot', () => {
		expect(render(<TextLink />).container.firstChild).toMatchSnapshot();
	});

	it('should allow as override', () => {
		expect(
			render(<TextLink as={<p />}>Test</TextLink>).container.firstChild,
		).toMatchSnapshot();
	});
});
