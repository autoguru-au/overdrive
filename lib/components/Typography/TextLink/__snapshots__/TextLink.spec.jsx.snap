// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextLink /> should allow as override 1`] = `
<p>
  <span
    class="root sizeScale3 strong root strong"
  >
    Test
  </span>
</p>
`;

exports[`<TextLink /> should match the snapshot 1`] = `
<a
  rel="noopener noreferrer"
>
  <span
    class="root sizeScale3 strong root strong"
  />
</a>
`;
